src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +238,320 @@ def inner(tree, path, root_path):
+    """
+    api_tree = get_api_tree_file_name(text=True)
+    api_tui_tree = get_api_tree_file_name(tui=True)
+    text_files = [api_tree, api_tui_tree]
+    for text_file in text_files:
+        api_objects = [
+            line.rstrip("\n") for line in open(text_file, "r", encoding="utf-8")
+        ]
+        tfidf_vectorizer = TfidfVectorizer()
+        tfidf_matrix = tfidf_vectorizer.fit_transform(api_objects)
+        for query in queries:
+            query_vector = tfidf_vectorizer.transform([query])
+            cosine_similarities = cosine_similarity(query_vector, tfidf_matrix)
+            if wildcard:
+                most_similar_api_object_index = cosine_similarities.argmax()
+                if query in api_objects[most_similar_api_object_index]:
+                    print(api_objects[most_similar_api_object_index])
+            else:
+                most_similar_api_object_indices = cosine_similarities.argsort()
+                for most_similar_api_object_index in reversed(
+                    most_similar_api_object_indices[0][-3:]
+                ):
+                    if query in api_objects[most_similar_api_object_index]:
+                        print(api_objects[most_similar_api_object_index])
+
+
+def _get_wildcard_matches_for_word_from_names(word: str, names: list):
+    """Get wildcard matches for the given word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
+    names: list
+        All API object names.
+
+    Returns
+    -------
+    wildcard_matches: list
+        Matched API object names.
+    """
+    wildcard_matches = [name for name in names if fnmatch.fnmatch(name, word)]
+    valid_wildcard_matches = [
+        wildcard_match for wildcard_match in wildcard_matches if wildcard_match in names
+    ]
+    return valid_wildcard_matches
+
+
+def _search_wildcard(search_string: str):
+    """Perform wildcard search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+
+    Returns
+    -------
+        List of search string matches.
+    """
+    names = _get_api_object_names()
+    queries = _get_wildcard_matches_for_word_from_names(search_string, names)
+    if queries:
+        _print_search_results(queries, wildcard=True)
+
+
+def _get_exact_match_for_word_from_names(
+    word: str,
+    names: list,
+):
+    """Get exact match for the given word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
+    names: list
+        All API object names.
+
+    Returns
+    -------
+    exact_matche: list
+        List of exact match.
+    """
+    return [name for name in names if word == name]
+
+
+def _get_close_matches_for_word_from_names(
+    word: str,
+    names: list,
+):
+    """Get close matches for the given word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
```

Comment:
```suggestion
        Word to search for.
```
I won't mark future occurrences but please do a search and replace to correct them all.

Reply:
Done. Thank you.